code/player/Player.Jobs.cs

```diff
@@ -7,11 +9,140 @@ public partial class Player
     [BindComponent]
     public Jobs.PlayerJob Job { get; }
 
+    //The default cooldown time for setting the forgiveness timer
+    float cooldownTime => 30.0f;
+
+    /// <summary>
+    /// The cooldown after leaving a job to prevent being assigned to a new job too quickly
+    /// </summary>
+    public TimeUntil JobCooldown { get; set; }
+
+    /// <summary>
+    /// Set the job to the player
+    /// </summary>
+    /// <param name="details">The details for that job to be assigned for the player</param>
     public void SetJob(Jobs.JobDetails details)
     {
         if (Game.IsClient) throw new System.Exception("Cannot set job on client!");
 
+        //Remove any current job and assign the new one
         Components.RemoveAny<Jobs.PlayerJob>();
         Components.Add(Jobs.PlayerJob.CreateFromDetails(details));
     }
+
+    //Job server ticking
+    [GameEvent.Tick.Server]
+    protected void JobServerTick()
+    {
+        //This could change if proper jobs have guest abilities
+        //if that happens, I'll think of a better check -ItsRifter
+
+        //This is a guest job
+        if (Job.HasAbility(JobAbilities.Guest)) return;
+
+        //Forgiveness timer hasn't finished
+        if (Job.JobDetails.ForgiveTimer > 0.0f) return;
+
+        //If there are recent fails and the timer has finished, forgive the employee for a fail
+        if(Job.JobDetails.Fails > 0)
+            SubtractJobFails();
```

Comment:
This should be refactored out into it's own function like
`HandleJobFailures()`
which does the whole kitten kaboodle of
- check if we even need to run this logic
- see if they have failures
- remove a failure if it's time